File: src/CSharp.Forms/UserControl1.cs
using CSharp.Repository.Interfaces;
using CSharp.Repository.Services;

namespace CSharp.Forms
{
    public partial class UserControl1 : UserControl
    {

        private readonly IHelloWorkRepository? _repository;

        public UserControl1()
        {
            if (ServiceLocator.ServiceProvider != null)
                _repository = ServiceLocator.GetService<IHelloWorkRepository>();
            
            InitializeComponent();
        }

        private void bnSayHello_Click(object sender, EventArgs e)
        {
            MessageBox.Show(_repository?.SayHello);
        }
    }
}

File: src/VB.Forms/UserControl1.vb
Imports VB.Repository.VB.Repository.Interfaces
Imports VB.Repository.VB.Repository.Services

Public Class UserControl1

    Private ReadOnly _repository As IHelloWorkRepository

    Public Sub New()
        If ServiceProvider IsNot Nothing Then
            _repository = ServiceLocator.GetService(Of IHelloWorkRepository)()
        End If
        InitializeComponent()
    End Sub


    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles bnSayHello.Click
        MessageBox.Show(_repository.SayHello)
    End Sub

End Class
